src/Controller/ResourceController.php

```diff
@@ -345,27 +364,41 @@ private function getDirectory(Request $request, $path, $responseFormat, array $v
             ];
         }
 
-        $content = $this->getGraphForPath($request, $path)->serialise($responseFormat, $options);
+        $content = $this->getGraphForPath($request, $requested_path)->serialise($responseFormat, $options);
 
         $headers = [
-            "Last-Modified" => $modifiedTime->format(\DateTime::W3C),
-            "Link" => ["<{$this->LDP_NS}Resource>; rel=\"type\"",
-                       "<{$this->LDP_NS}BasicContainer>; rel=\"type\""],
-            "Vary" => "Accept",
+            "Link" => ["<".self::LDP_NS."Resource>; rel=\"type\"",
+                       "<".self::LDP_NS."BasicContainer>; rel=\"type\""],
             "Content-Type" => $responseMimeType,
             "Content-Length" => strlen($content)
         ];
 
-        if (!$doGet) {
-            $content = '';
+        if ($request->getMethod() == 'GET') {
+            $response->setContent($content);
         }
-        return new Response($content, 200, $headers);
+        $response->headers->add($headers);
+        return $response;
+    }
+
+    /**
+     * Simple Convulsive Etag generator for files
+     *
+     * @return array
+     */
+    private function provideEtag()
```

Comment:
Rather than having this function operate on mutable (and possibly `null` member fields), I'd rather see this method be more like a static function that operates on an input string. That makes the function considerably more flexible (I'm thinking specifically of its use with RDF content, where the serialization will influence the ETag). Something like this:

```
private function generateEtag($input)
{
    return sha1($input);
}
```

One could argue that the new function is longer than just using `sha1` directly (which would also be fine, if you'd like to go that route), but it is more semantically meaningful.